src/libraries/System.Text.Json/Common/JsonSimpleNamingPolicy.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSimpleNamingPolicy : JsonNamingPolicy
+    {
+        private readonly bool _lowercase;
+        private readonly char _boundary;
+
+        internal JsonSimpleNamingPolicy(bool lowercase, char boundary) =>
+            (_lowercase, _boundary) = (lowercase, boundary);
+
+        public override string ConvertName(string name)
+        {
+            int bufferLength = name.Length * 2;
+            char[]? buffer = bufferLength > 512
+                ? ArrayPool<char>.Shared.Rent(bufferLength)
+                : null;
+
+            int resultLength = 0;
+            Span<char> result = buffer is null
+                ? stackalloc char[512]
+                : buffer;
+
+            void WriteWord(ref Span<char> result, ReadOnlySpan<char> word)
```

Comment:
`ref` not needed
```suggestion
            void WriteWord(Span<char> result, ReadOnlySpan<char> word)
```

Reply:
Checked it carefully, actually it's needed because `result` can be modified inside `WriteWord` by `ExtendBuffer`. Currently, there are no test cases which can cause an overflow, but that's because of the initial buffer length which is twice as larger than an input. I plan to change that part and use only a 20% longer buffer that an input to avoid potential memory allocations.

I will port my latest changes which doesn't include the thing I mentioned, but here are intermediate results for what I did in my package ([implementation](https://github.com/YohDeadfall/Yoh.Text.Json.NamingPolicies/blob/d8ca04c0b23e5cb728f4a70a11d82f65c9a65b49/src/JsonSeparatorNamingPolicy.cs)):

```
BenchmarkDotNet=v0.13.2, OS=Windows 11 (10.0.25126.1000)
AMD Ryzen 7 4700U with Radeon Graphics, 1 CPU, 8 logical and 8 physical cores
.NET SDK=6.0.300
  [Host]     : .NET 6.0.5 (6.0.522.21309), X64 RyuJIT AVX2
  DefaultJob : .NET 6.0.5 (6.0.522.21309), X64 RyuJIT AVX2


|         Method |                Name |       Mean |    Error |   StdDev |   Gen0 | Allocated |
|--------------- |-------------------- |-----------:|---------:|---------:|-------:|----------:|
| SnakeLowerCase | ATow(...)ough [716] | 6,300.9 ns |  9.33 ns |  8.27 ns | 0.8392 |    1776 B |
| SnakeUpperCase | ATow(...)ough [716] | 6,415.5 ns | 55.13 ns | 51.57 ns | 0.8469 |    1776 B |
| KebabLowerCase | ATow(...)ough [716] | 6,359.3 ns | 16.05 ns | 14.22 ns | 0.8469 |    1776 B |
| KebabUpperCase | ATow(...)ough [716] | 6,376.0 ns | 17.22 ns | 15.27 ns | 0.8469 |    1776 B |
| SnakeLowerCase |      XMLHttpRequest |   119.9 ns |  0.92 ns |  0.86 ns | 0.0267 |      56 B |
| SnakeUpperCase |      XMLHttpRequest |   122.7 ns |  1.03 ns |  0.97 ns | 0.0267 |      56 B |
| KebabLowerCase |      XMLHttpRequest |   119.1 ns |  0.93 ns |  0.87 ns | 0.0267 |      56 B |
| KebabUpperCase |      XMLHttpRequest |   120.9 ns |  0.77 ns |  0.68 ns | 0.0267 |      56 B |

|         Method |                Name |       Mean |    Error |   StdDev |   Gen0 | Allocated |
|--------------- |-------------------- |-----------:|---------:|---------:|-------:|----------:|
| SnakeLowerCase | ATow(...)ough [716] | 5,863.4 ns | 12.92 ns | 11.46 ns | 0.8392 |    1776 B |
| SnakeUpperCase | ATow(...)ough [716] | 5,814.0 ns | 39.84 ns | 37.27 ns | 0.8469 |    1776 B |
| KebabLowerCase | ATow(...)ough [716] | 5,937.5 ns | 18.65 ns | 16.53 ns | 0.8469 |    1776 B |
| KebabUpperCase | ATow(...)ough [716] | 5,864.1 ns | 29.58 ns | 26.22 ns | 0.8469 |    1776 B |
| SnakeLowerCase |      XMLHttpRequest |   118.7 ns |  2.37 ns |  2.22 ns | 0.0267 |      56 B |
| SnakeUpperCase |      XMLHttpRequest |   116.5 ns |  0.62 ns |  0.52 ns | 0.0267 |      56 B |
| KebabLowerCase |      XMLHttpRequest |   118.8 ns |  0.94 ns |  0.88 ns | 0.0267 |      56 B |
| KebabUpperCase |      XMLHttpRequest |   118.2 ns |  0.81 ns |  0.72 ns | 0.0267 |      56 B |
```